spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -30,8 +30,11 @@ protected String metadataTableName(MetadataTableType type) {
     String tableName = table().toString();
     if (tableName.contains("/")) {
       return tableName + "#" + type;
-    } else if (tableName.startsWith("hadoop.") || tableName.startsWith("hive.")) {
-      // HiveCatalog and HadoopCatalog prepend a logical name which we need to drop for Spark 2.4
+    } else if (tableName.startsWith("hadoop.")) {
+      // Load a path by HadoopCatalog or HadoopTables
```

Comment:
How about "for HadoopCatalog tables, use the table location to load the metadata table because IcebergCatalog uses HiveCatalog when the table is identified by name".